### ISSUE TITLE
Add "Screen" blend mode (DEF-1182)

### ISSUE THREAD

**Issue by britzl** [defold/defold#3419]
It has been requested that we add a "Screen" blend mode in addition to "Alpha", "Add" and "Multiply".

See: (LINK REMOVED)
2016-01-18 15:51:28 (sven.andersson)
Just need a new OGL blend mode; (LINK REMOVED)
2017-09-15 13:02:31 (Fredrik.Malmer)
Scroll down to "Fetch Framebuffer Data for Programmable Blending"
(LINK REMOVED)Will work on more or less all mobile devices, but not on desktop afaik.

**Comment by beldenfox** [comment]
This doesn't require frame buffer fetch and works fine on my desktop. In the end this is just like all the existing blend modes but with `SourceBlendFactor = ONE_MINUS_DST_COLOR, DestinationBlendFactor = ONE` (or `SourceBlendFactor = ONE, DestinationBlendFactor = ONE_MINUS_SRC_COLOR`). Math available on request.
Since this is just a variant of the existing blend modes it's pretty easy to track down where changes would need to be made in the engine. Are there also tests that would need to be updated?

**Comment by britzl** [comment]
@beldenfox Cool! Have you already started on the required changes? Do you need some help to get started? Please join the #source-code channel on Discord if you have general questions regarding the source code or building the engine or editor.

**Comment by beldenfox** [comment]
I can submit a PR on this if you're interested. I've only updated the core engine code and did a tiny tweak to the editor. Are there any tests or documentation you want me to update as part of the PR?

I just added Screen everywhere Multiply was already available even though I'm not sure it's useful for, say, a Spine. And it seems that changing the blend mode on a text label doesn't have any effect, even the existing Multiply blend mode is ignored. Is that a known issue?
